.github/workflows/multi_arch_build_portable_linux.yml

```
@@ -0,0 +1,682 @@
+#
+# This workflow builds TheRock in stages:
+# 1. foundation (generic) - sysdeps, base
+# 2. compiler-runtime (generic) - compiler, runtimes, profiler-core
+# 3. math-libs (per-arch) - BLAS, FFT, etc.
+# 4. comm-libs (per-arch) - RCCL (parallel to math-libs)
+# 5. dctools-core (generic) - RDC (parallel to math-libs)
+# 6. profiler-apps (generic) - rocprofiler-systems (parallel to math-libs)
+#
+# Artifacts flow between stages via S3 using the artifact_manager.py tool.
+
+name: Multi-Arch Build (Linux)
+
+on:
+  workflow_call:
+    inputs:
+      amdgpu_families_json:
+        type: string
+        required: true
+        description: "JSON array of GPU families for per-arch stages"
+      artifact_group:
+        type: string
+        required: true
+        description: "Artifact group identifier"
+      build_variant:
+        type: string
+        required: true
+        description: "Build variant (release, asan)"
+      package_version:
+        type: string
+        required: true
+        description: "Package version string"
+      enabled_stages:
+        type: string
+        required: true
+        description: "Comma-separated list of enabled stages"
+
+permissions:
+  contents: read
+
+env:
+  CONTAINER_IMAGE: ghcr.io/rocm/therock_build_manylinux_x86_64@sha256:4af52d56d91ef6ef8b7d0a13c6115af1ab2c9bf4a8a85d9267b489ecb737ed25
+  CCACHE_CONFIGPATH: ${{ github.workspace }}/.ccache/ccache.conf
+  CACHE_DIR: ${{ github.workspace }}/.container-cache
+  TEATIME_FORCE_INTERACTIVE: 0
+
+jobs:
+  # ==========================================================================
+  # STAGE: foundation (generic)
+  # ==========================================================================
+  foundation:
+    name: Stage - Foundation
+    if: contains(inputs.enabled_stages, 'foundation')
+    runs-on: azure-linux-scale-rocm
+    timeout-minutes: 180  # 3 hours
+    permissions:
+      id-token: write
+    container:
+      image: ghcr.io/rocm/therock_build_manylinux_x86_64@sha256:4af52d56d91ef6ef8b7d0a13c6115af1ab2c9bf4a8a85d9267b489ecb737ed25
+      options: -v /runner/config:/home/awsconfig/
+    env:
+      AWS_SHARED_CREDENTIALS_FILE: /home/awsconfig/credentials.ini
+      STAGE_NAME: foundation
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@93cb6efe18208431cddfb8368fd83d5badbf9bfd # v5.0.1
+
+      - name: Install python deps
+        run: pip install -r requirements.txt
+
+      - name: Adjust git config
+        run: |
+          git config --global --add safe.directory $PWD
+          git config fetch.parallel 10
+
+      - name: Setup ccache
+        run: |
+          ./build_tools/setup_ccache.py \
+            --config-preset "github-oss-presubmit" \
+            --dir "$(dirname $CCACHE_CONFIGPATH)" \
+            --local-path "$CACHE_DIR/ccache"
+
+      - name: Fetch sources
+        timeout-minutes: 30
+        run: ./build_tools/fetch_sources.py --stage ${STAGE_NAME} --jobs 12
```

Let's run `health_status.py` in here too, like so: https://github.com/ROCm/TheRock/blob/d4186d98752ef3e094d1a682ff3d7ac20935e286/.github/workflows/build_portable_linux_artifacts.yml#L102-L104

That reports disk usage, CPU specs, and other machine information.
```suggestion
      - name: Setup ccache
        run: |
          ./build_tools/setup_ccache.py \
            --config-preset "github-oss-presubmit" \
            --dir "$(dirname $CCACHE_CONFIGPATH)" \
            --local-path "$CACHE_DIR/ccache"

      - name: Runner health status
        run: |
          ./build_tools/health_status.py

      - name: Fetch sources
        timeout-minutes: 30
        run: ./build_tools/fetch_sources.py --stage ${STAGE_NAME} --jobs 12
```